internal/objectbuilder/clusterclass.go

```diff
@@ -0,0 +1,380 @@
+package objectbuilder
```

Comment:
If I got this right from https://github.com/kubernetes-sigs/cluster-api/pull/5040#issuecomment-891084020 and from (https://github.com/kubernetes-sigs/cluster-api/pull/5040#discussion_r680128712), the intent is to keep everything in a the test types package for now, but splitting things in different files.
@sbueringer to add on this

Reply:
Right - that makes a lot of sense I think. I'll update the structure to put everything under testtypes.